common-custom-user-data-maven-extension/src/main/java/com/gradle/BuildCacheApiAccessor.java

```diff
@@ -0,0 +1,61 @@
+package com.gradle;
+
+import com.gradle.maven.extension.api.cache.BuildCacheApi;
+import org.apache.maven.MavenExecutionException;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
+import org.codehaus.plexus.component.repository.exception.ComponentLookupException;
+
+import java.util.Optional;
+
+import static java.util.Comparator.comparing;
+
+final class BuildCacheApiAccessor {
```

Comment:
This is largely copied fro `BuildScanApiAccessor`, right? Can we replace it with a single static parameterized method `<T> T lookup(Class<T> componentClass, PlexusContainer container, Class<?> extensionClass)` in a `ComponentAccessor` class?

Reply:
Done.